### PR TITLE
Cleanup clang warnings and move to new version of dependencies

### PR DESCRIPTION
**Pull request by nalinigans**
Checking for `bcf_hdr_sync()` return values to silence clang, just logging the errors for now. Should we throw an exception here?

### CODE REVIEW

File: src/main/cpp/include/genomicsdb/genomicsdb_iterators.h [hunk 1]
```diff
@@ -340,7 +341,6 @@ class SingleCellTileDBIterator {
   //Flag that specifies if this is the first time reading from TileDB
   bool m_first_read_from_TileDB;
   unsigned m_END_query_idx;
-  const VariantArraySchema* m_variant_array_schema;
```
Comment (kgururaj) [inline src/main/cpp/include/genomicsdb/genomicsdb_iterators.h:343]: The GVCF iterator gets dim_domains from this object - we can either store the dim domains or just keep this object.

File: src/main/cpp/src/genomicsdb/genomicsdb_iterators.cc [hunk 2]
```diff
@@ -43,7 +44,7 @@ SingleCellTileDBIterator::SingleCellTileDBIterator(TileDB_CTX* tiledb_ctx,
     const TileDB_Array* tiledb_array,
     const VidMapper* vid_mapper, const VariantArraySchema& variant_array_schema,
     const std::string& array_path, const VariantQueryConfig& query_config, const size_t buffer_size)
-  : m_variant_array_schema(&variant_array_schema), m_query_config(&query_config),
```
Comment (kgururaj) [inline src/main/cpp/src/genomicsdb/genomicsdb_iterators.cc:46]: Same comment as before

File: src/main/cpp/src/query_operations/broad_combined_gvcf.cc [hunk 3]
```diff
@@ -308,7 +310,9 @@ BroadCombinedGVCFOperator::BroadCombinedGVCFOperator(VCFAdapter& vcf_adapter, co
                                          +callset_name+" to the combined VCF/gVCF header");
     }
   }
-  bcf_hdr_sync(m_vcf_hdr);
+  if (bcf_hdr_sync(m_vcf_hdr)) {
+    logger.error("bcf_hdr_sync() failed while {}:{}", "adding samples to template header", __LINE__);
+  }
```
Comment (kgururaj) [inline src/main/cpp/src/query_operations/broad_combined_gvcf.cc:315]: A quick look at bcf_hdr_sync shows that this error can occur only if there's a memory allocation failure - so maybe a short phrase about realloc() failed and throw an exception.

File: src/main/cpp/src/vcf/vcf2binary.cc [hunk 4]
```diff
@@ -1121,7 +1121,9 @@ void VCF2Binary::write_partition_data(File2TileDBBinaryColumnPartitionBase& part
   //The second parameter is useful if the file handler is open, but the buffer was full in a previous call
   auto has_data = seek_and_fetch_position(partition_info, is_read_buffer_exhausted, m_close_file, false);
   while (has_data) {
-    bcf_write(vcf_partition.m_split_output_fptr, hdr, vcf_reader_ptr->get_line());
+    if (bcf_write(vcf_partition.m_split_output_fptr, hdr, vcf_reader_ptr->get_line())) {
+      logger.fatal(VCF2BinaryException(logger.format("Error writing VCF data for partition at {}", __LINE__)));
```
Comment (kgururaj) [inline src/main/cpp/src/vcf/vcf2binary.cc:1125]: Yeah, exception makes sense here

**Review by kgururaj — APPROVED** [review]
Minor changes requested.

File: src/main/cpp/include/genomicsdb/genomicsdb_iterators.h (hunk 1 above)
Comment (nalinigans) [reply src/main/cpp/include/genomicsdb/genomicsdb_iterators.h:343]: We can keep this object. Thanks.

**Comment by nalinigans** [comment]
Made changes as requested in review. Catching error status from bcf_hdr_sync() and bcf_write() and throwing proper GenomicsDB exceptions might help with the issues in #108 and https://github.com/broadinstitute/gatk/issues/6745 as the stack trace shows errors emanating from `org.genomicsdb.reader.GenomicsDBFeatureReader.generateHeadersForQuery(GenomicsDBFeatureReader.java:200)`

```java.io.IOException: GenomicsDB JNI Error: std::exception
	at org.genomicsdb.reader.GenomicsDBQueryStream.jniGenomicsDBInit(Native Method)
	at org.genomicsdb.reader.GenomicsDBQueryStream.<init>(GenomicsDBQueryStream.java:209)
	at org.genomicsdb.reader.GenomicsDBQueryStream.<init>(GenomicsDBQueryStream.java:182)
	at org.genomicsdb.reader.GenomicsDBQueryStream.<init>(GenomicsDBQueryStream.java:91)
	at org.genomicsdb.reader.GenomicsDBFeatureReader.generateHeadersForQuery(GenomicsDBFeatureReader.java:200)
	at org.genomicsdb.reader.GenomicsDBFeatureReader.<init>(GenomicsDBFeatureReader.java:85)
	at org.broadinstitute.hellbender.tools.genomicsdb.GenomicsDBImportIntegrationTest.getGenomicsDBFeatureReader(GenomicsDBImportIntegrationTest.java:927)
```

**Review by kgururaj — APPROVED** [review]

**Pull request merged**